### PR TITLE
Add default values to tokenized strings

### PR DESCRIPTION
**Pull request by mikhail**
@diranged @swaggy

### CODE REVIEW

**Comment by swaggy** [comment]
LGTM once we agree on a better syntax than `?`

File: kingpin/test/test_utils.py [hunk 1]
```diff
@@ -33,6 +33,20 @@ def test_populate_with_env(self):
         result = utils.populate_with_tokens(string, tokens)
         self.assertEquals(result, expect)
 
+    def test_populate_with_values_not_default(self):
+        tokens = {'UNIT_TEST': 'FOOBAR', 'SECOND_UNIT': 'BARBAR'}
+        string = 'Unit %UNIT_TEST?DEFAULT% Test %SECOND_UNIT?DEFAULT2%'
```
Comment (diranged) [inline kingpin/test/test_utils.py]: I'd rather see us use `|` .. which means "or".. Thoughts"

Comment (mikhail) [inline kingpin/test/test_utils.py]: We've been discussing that for a long time... I don't really care, and i'm even ok with allowing multiple values. could be `|` or double: `||` could even be "space or space" ` or ` `?=` `:` ... don't know

Comment (diranged) [inline kingpin/test/test_utils.py]: I suggest just a single `|` ... using `?=` is too bash-like. A semi-colon would be OK too I suppose, but I feel like `|` really means `or` to most people.

**Pull request merged**